src/Native/Runtime/GCHelpers.cpp

```diff
@@ -123,3 +123,100 @@ COOP_PINVOKE_HELPER(Int64, RhGetLastGCDuration, (Int32 generation))
 {
     return GCHeap::GetGCHeap()->GetLastGCDuration(generation);
 }
+
+#ifdef WRITE_BARRIER_CHECK
+extern uint8_t* g_GCShadow;
+extern uint8_t* g_GCShadowEnd;
+extern uint8_t* g_lowest_address;
+extern uint8_t* g_highest_address;
+#endif
+
+extern uint32_t* g_card_table;
+static UInt32 INVALIDGCVALUE = 0xcccccccd;
+
+extern "C" void RhpBulkWriteBarrier(void* pMemStart, UInt32 cbMemSize)
```

Comment:
We should split this into forceinline  worker method - that can be used in places like RhBulkMoveWithWriteBarrier where we would really love to avoid the call overhead; and the extern "C" entrypoint - that just calls the forceinline worker method.